### ISSUE TITLE
clarify queryables

### ISSUE THREAD

**Issue by tomkralidis** [opengeospatial/ogcapi-records#33]
In CSW we have the concept of core queryables which were a set of specified properties that client and server implementations could use for querying with consistency, e.g.:

- `dc:title`
- `csw:AnyText`

Thereafter, application profiles (e.g. APISO) defined their own queryables, e.g.:
- `apiso:Title`
- `apiso:TopicCategory`

The idea here is that a server implementation then could map their underlying implementation (e.g. RDBMS, search engine) to the standard queryables to support cross-catalogue query interoperability.

For OGC API - Records core, we could/will have a `/collections/{collectionId}/queryables` endpoint.  Options for advertising queryables:
1. each record collection has its own queryables at the discretion of the server implementation
2. each record collection must implement (at least) a defined set of core queryables (i.e. title, abstract, keywords, etc.) as defined by OGC API - Records
3. each record collection must implement (at least) queryables as per the [core record model](https://github.com/opengeospatial/ogcapi-records/blob/master/core/code/record.json)
4. other options?

**Comment by tomkralidis** [comment]
In addition, can/should queryables be additionally advertised at the `/collections/{collectionId}` endpont, or only at `/collections/{collectionId}/queryables` ?

**Comment by pvretano** [comment]
In my prototype implementation I advertise them at both the /collections/{collectionId} endpoint (in a queryables section) and at the /collections/{collectionId/queryables endpoint. 

- http://www.pvretano.com/cubewerx/cubeserv/default/wrs/4.0/collections/sentinel1cat?f=json
- http://www.pvretano.com/cubewerx/cubeserv/default/wrs/4.0/collections/sentinel1cat/queryables?f=json

However, I really think that only the /collections/{collectionId}/queryables endpoint is necessary.  There is no point in cluttering the collectionId endpoint with more and more information.  The most I would say is to include a link the collectionId endpoint (rel=queryables or somethings like that) with the location where you can get the queryables.

**Comment by pvretano** [comment]
I would make the "core" queryables a very short list and allow a server to advertise any number of additional queryables as desired or necessary for a specific application.  As a starting point, I would proposed that the following be the "core" set of queryables: dc:identifier, dc:title, dc:type, dc:subject, dc:format, dc:relation, dct:modified, dct:abstract, dct:spatial, ows:BoundingBox and csw30:TemporalExent.  Please note that the namespace prefixes are only used here to tie these proposed queryables back to CSW30 and the Dublin core metadata element from where the list was derived.  This is just a starting point of course...

**Comment by pvretano** [comment]
One final point, we should remove all the references to "queryables" in the specification and simple refer to the CQL specification which defines the endpoint and the response schema.  Best not redefine the wheel where!

**Comment by tomkralidis** [comment]
Thanks Peter. I pushed some updates into #35 to this effect.

Having said this:
- do we define OARec core queryables through the CQL extension proper?  Or do we identify the core queryables in the 'core OARec spec "as implemented through the standard CQL extension" ?  (not sure if this is clear)?
- should `dc:identifier` be a queryable?  What about/how does this compare to `/collections/{collectionID}/items/{itemId}` ?

**Comment by tomkralidis** [comment]
Meeting 2020-05-04:

- only required in `/collections/{collectionId}/queryables`
- reference CQL
- list mandatory queryables
- include identifier as part of core queryables

**Comment by pvretano** [comment]
Meeting 2020-05-18:

- The currently proposed record shema is based on GeoJSON (see [https://github.com/opengeospatial/ogcapi-records/blob/master/core/examples/json/record.json](here))
- GeoJSON already defines a set of keys that includes a "properties" key. 
- The discussion at today's teleconference focused on which of the proposed keys that go into the "properties" object should be mandatory, recommended or dropped.
- The proposed list is: _title, description, language, issued, modified, publisher, keywords, themes, contactPoint, accrualPeriodicity, langingPage, license, rights, accessURL, downloadURL, formats, bytSize, homePage_
- We managed to discuss up to the _landingPage_ property and we decided the following:

**MANDATORY**: _title, description, keywords_
**RECOMMENDED**: _modified, publisher, themes (renamed to category), contactPoint_
**DROPPED**: _issued, accrualPeriodicity, landingPage_

- Next meeting: finish discussion about queryables and review the CQL link query extension

**Comment by uvoges** [comment]
Not to forget: For the harvesting use case we need a "last modified date of the metadata" (which may be the same as the "last modified date of the data")

**Comment by pvretano** [comment]
SWG Meeting Notes: 01-JUN-2020:

**MANDATORY**: _id, title, description, keywords_
**RECOMMENDED**: _modified, publisher, themes (renamed to category), contactPoint_,(license,rights,accessRights -- see DCAT for a full discussion of the distinctions and how to use these parameters, see also 19115, see also https://github.com/qgis/QGIS/blob/master/resources/qgis-resource-metadata.xsd#L45-L54), formats (verify that this is like mediaType)
**DROPPED**: _issued, accrualPeriodicity, landingPage_

- accessURL, downloadURL, homePage - converging on adding a section to the core queryables where we can list these types of links BUT not put them in the links section where we have links mostly for hypermedia controls (i.e. navigating resources within the API).

- we have two types of links (1) links that are related to the catalogue record itself and are used to navigate within the catalogue API; these are hypermedia control type links and they go into the "links" section (2) links that are related to the thing that the catalogue record is describing; we need a different "links" section for these links and we need to think a little more about what we call this section

**Comment by pvretano** [comment]
SWG Meeting Notes: 16-JUN-2020
(1) Discussion about binding section
(2) Make registry/repository separate user cases (see ISO 19135)
(3) Mention ranking in sorting (SEO use case) (see issue #37 )

**Comment by tomkralidis** [comment]
>> accessURL, downloadURL, homePage - converging on adding a section to the core queryables where we can list these types of links BUT not put them in the links section where we have links mostly for hypermedia controls (i.e. navigating resources within the API).

What about grouping these into a section called `distribution` either as keys or a list or link objects?
```json
        "distribution": {
            "access": {
                "rel": "items",
                "type": "image/png",
                "title": "OGC:WMS map of this dataset",
                "href": "https://example.org/wms/service=WMS&version=1.3.0&request=GetMap&layer={layer}&bbox={miny},{minx},{maxy},{maxx}&format={format}&crs={crs}&width={width}&height={height}",
                "templated": true
            },
            "download": {
                "rel": "items",
                "type": "image/png",
                "title": "static download of data",
                "href": "https://example.org/dataset.tgz",
                "templated": true
            },
            "homepage": {
                "rel": "about",
                "type": "text/html",
                "title": "dataset page",
                "href": "https://example.org/dataset"
            }
        }
```

**Comment by mhogeweg** [comment]
is this still about queryables? seems these links are more related to a response

**Comment by pvretano** [comment]
@mhogeweg I believe it is.
@tomkralidis we already discussed this issue in the SWG and have already resolved it as described in the fragment that you cite in your comment above.  We added another **core queryable** named _"associations"_ that is an array of links, like the links section, to associated resources like bulk download packages, additional descriptive metadata, etc. that are outside the OGC API sphere.  Personally, I think this distinction is a little weak and I would be happy if we simply stuck with the links sections for both purposes.  Alternatively we need to be much clearer about what kinds of links go into each array.
